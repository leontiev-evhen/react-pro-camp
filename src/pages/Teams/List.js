import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Row, Col, Figure } from 'react-bootstrap';
import { ErrorBoundary } from '../../components';
import './style.css';

function List({ teams }) {
	let i,
		j,
		list = [],
		tempArray = [],
		chunk = 4;

	for (i = 0, j = teams.length; i < j; i += chunk) {
		tempArray = teams.slice(i, i + chunk);
		let items = [];

		tempArray.map(item => {
			return items.push(
				<ErrorBoundary key={item.team_id}>
					<Col md="3">
						<div className="section-team">
							<Link to={`/team/${item.team_id}`}>
								<div className="section-team-title">{item.name}</div>
								<Figure>
									<Figure.Image alt="logo" src={item.logo} />
								</Figure>
								<small>{item.venue_city}</small>
							</Link>
						</div>
					</Col>
				</ErrorBoundary>
			);
		});
		list.push(
			<div key={i} className="my-3">
				<Row>{items}</Row>
			</div>
		);
	}
	return <ul>{list}</ul>;
}

List.propTypes = {
	teams: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default List;
