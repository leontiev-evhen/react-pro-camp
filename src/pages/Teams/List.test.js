import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List', () => {
	const props = {
		teams: [
			{
				team_id: 1,
				name: 'team1',
				logo: 'logo1.png',
			},
			{
				team_id: 2,
				name: 'team2',
				logo: 'logo2.png',
			},
			{
				team_id: 3,
				name: 'team3',
				logo: 'logo3.png',
			},
		],
	};
	const Component = shallow(<List {...props} />);

	it('List component render correctly', () => {
		expect(Component).toMatchSnapshot();
	});

	it('Renders 3 teams', () => {
		expect(Component.find('.section-team')).toHaveLength(3);
	});

	describe('Correct render for first team', () => {
		it('Team name', () => {
			expect(
				Component.find('.section-team-title')
					.first()
					.text()
			).toEqual(props.teams[0].name);
		});

		it('Team logo', () => {
			expect(
				Component.find('.section-team')
					.first()
					.find('FigureImage')
					.prop('src')
			).toEqual(props.teams[0].logo);
		});
	});
});
