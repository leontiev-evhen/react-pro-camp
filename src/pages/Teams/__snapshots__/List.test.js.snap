// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List List component render correctly 1`] = `
<Bootstrap(Row)>
  <ErrorBoundary
    key="1"
  >
    <Col
      md="3"
    >
      <div
        className="section-team center"
      >
        <Link
          to="/team/1"
        >
          <div
            className="section-team-title"
          >
            team1
          </div>
          <Figure>
            <FigureImage
              alt="logo"
              fluid={true}
              src="logo1.png"
            />
          </Figure>
          <small />
        </Link>
      </div>
    </Col>
  </ErrorBoundary>
  <ErrorBoundary
    key="2"
  >
    <Col
      md="3"
    >
      <div
        className="section-team center"
      >
        <Link
          to="/team/2"
        >
          <div
            className="section-team-title"
          >
            team2
          </div>
          <Figure>
            <FigureImage
              alt="logo"
              fluid={true}
              src="logo2.png"
            />
          </Figure>
          <small />
        </Link>
      </div>
    </Col>
  </ErrorBoundary>
  <ErrorBoundary
    key="3"
  >
    <Col
      md="3"
    >
      <div
        className="section-team center"
      >
        <Link
          to="/team/3"
        >
          <div
            className="section-team-title"
          >
            team3
          </div>
          <Figure>
            <FigureImage
              alt="logo"
              fluid={true}
              src="logo3.png"
            />
          </Figure>
          <small />
        </Link>
      </div>
    </Col>
  </ErrorBoundary>
</Bootstrap(Row)>
`;
